main/src/ca/uwaterloo/flix/language/phase/typer/ConstraintGen.scala

```diff
@@ -102,13 +102,6 @@ object ConstraintGen {
         // - have better performance (we don't generate unnecessary type variables)
         //
         val knownTarget = exp match {
-          case KindedAst.Expr.Def(sym, tvar1, loc1) =>
```

Comment:
I wonder... do we even need Expr.Def anymore?

Reply:
we do not no, pretty cool

Reply:
Very cool. Remove it.

Reply:
Yeah really cool. I definitely will :+1: